source/4.0/learn/getting-started/core-concepts.html.md

```diff
@@ -0,0 +1,189 @@
+---
+chapter: Getting Started
+title: Core Concepts
+---
+
+$TOC
+  1. [Reading Data](#reading-data)
+  2. [Writing Data](#writing-data)
+  3. [Repositories](#repositories)
+  4. [Relations](#relations)
+  5. [Commands](#commands)
+  6. [Mappers](#relation-mappers)    
+  7. [Changesets](#changesets)
+  8. [Adapters](#adapters)  
+$TOC
+
+Before you start writing any data access code with ROM, it's a good idea to
+get a high level overview of what each major component is and its overall 
+purpose.
+
+ROM fundamentally is a series of abstractions built on one another to create
+a very flexible system for accessing and manipulating complex data in your
+applications.
+
+The following diagram shows an overview of the ROM architecture and provides an
+outline of how data flows within a ROM based data access layer. Further down
+is a basic explanation of each component.
+
+![ROM Design Overview](images/rom-overview.jpg)
+
+## Data flow
+
+### Reading Data
+
+**Step 1:** Application Domain has a need for a particular entity/entities so it
+            calls into the appropriate repository and asks for an entity
+        
+**Step 2.** The Repository accepts the request and combines all of the
+            related associations together to create a datastore specific query
+            for the data
+        
+**Step 3.** Once a relation has been composed of all needed restrictions or 
+            selections, the relations `#each` method is called which triggers
+            the <mark>dataset</mark> to fetch the data.
+
+**Step 4.** Raw data is collected from the datastore and passed up to the relation.
+
+**Step 5.** The relation, if mappers are configured, maps the raw data into an
+            output entity.
+
+**Step 6.** The repository passes the entity on through to the Application Domain.
+
+### Writing Data
+
+**Step 7.** Application Domain requests an entity to be updated.
```

Comment:
This isn't accurate, because you don't really update entities in rom. What you do, is applying a command to a specific relation. It can create one or more tuples, update one or more tuples or delete one or more tuples. So ie you don't really need to request some entity in order to update it. In fact, we discourage this approach, as it needs additional overhead related to reading "entities".

Reply:
It's still inaccurate but I think my wording is causing a bit of confusion. By "requests an entity be updated" I really mean the application domain calls a method on a repository and passes in an entity (and by entity I mean either a custom object or rom struct or hash, whatever the app is using) that needs to be saved. Either way, I definitely need to make that more clear.

Any suggestions on alternate wording?